test/functional/mempool_accept.py

```diff
@@ -58,38 +56,32 @@ def check_mempool_result(self, result_expected, *args, **kwargs):
 
     def run_test(self):
         node = self.nodes[0]
+        miniwallet = MiniWallet(node)
 
         self.log.info('Start with empty mempool, and 200 blocks')
         self.mempool_size = 0
+        miniwallet.generate(10)
+        node.generate(190)
         assert_equal(node.getblockcount(), 200)
         assert_equal(node.getmempoolinfo()['size'], self.mempool_size)
-        coins = node.listunspent()
 
         self.log.info('Should not accept garbage to testmempoolaccept')
         assert_raises_rpc_error(-3, 'Expected type array, got string', lambda: node.testmempoolaccept(rawtxs='ff00baar'))
         assert_raises_rpc_error(-8, 'Array must contain exactly one raw transaction for now', lambda: node.testmempoolaccept(rawtxs=['ff00baar', 'ff22']))
         assert_raises_rpc_error(-22, 'TX decode failed', lambda: node.testmempoolaccept(rawtxs=['ff00baar']))
 
         self.log.info('A transaction already in the blockchain')
-        coin = coins.pop()  # Pick a random coin(base) to spend
-        raw_tx_in_block = node.signrawtransactionwithwallet(node.createrawtransaction(
-            inputs=[{'txid': coin['txid'], 'vout': coin['vout']}],
-            outputs=[{node.getnewaddress(): 0.3}, {node.getnewaddress(): 49}],
-        ))['hex']
-        txid_in_block = node.sendrawtransaction(hexstring=raw_tx_in_block, maxfeerate=0)
-        node.generate(1)
+        tx_in_block = miniwallet.send_self_transfer(from_node = node, default_fee = True)
+        node.generate(1) # Generate a block, tx gets mined therefore mempool is empty
         self.mempool_size = 0
         self.check_mempool_result(
-            result_expected=[{'txid': txid_in_block, 'allowed': False, 'reject-reason': 'txn-already-known'}],
-            rawtxs=[raw_tx_in_block],
+            result_expected=[{'txid': tx_in_block['txid'], 'allowed': False, 'reject-reason': 'txn-already-known'}],
+            rawtxs=[tx_in_block['hex']],
         )
 
         self.log.info('A transaction not in the mempool')
-        fee = Decimal('0.000007')
-        raw_tx_0 = node.signrawtransactionwithwallet(node.createrawtransaction(
-            inputs=[{"txid": txid_in_block, "vout": 0, "sequence": BIP125_SEQUENCE_NUMBER}],  # RBF is used later
-            outputs=[{node.getnewaddress(): Decimal('0.3') - fee}],
-        ))['hex']
+        fee = Decimal('0.00000672')
```

Comment:
You're changing the fee?

Reply:
Yea cause the previous fee of `fee = Decimal('0.000007')` was too low to pass the mempool test. I manually calculated the fee here `fee_rate = Decimal('0.00007672') - fee,`, to get the best option